core/templates/components/forms/schema-based-editors/schema-based-choices-editor.component.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for a schema-based editor for multiple choice.
+ */
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import { NG_VALUE_ACCESSOR, NG_VALIDATORS, ControlValueAccessor, Validator, AbstractControl, ValidationErrors } from '@angular/forms';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'schema-based-choices-editor',
+  templateUrl: './schema-based-choices-editor.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SchemaBasedChoicesEditorComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => SchemaBasedChoicesEditorComponent),
+      multi: true
+    },
+  ]
+})
+export class SchemaBasedChoicesEditorComponent
+implements ControlValueAccessor, OnInit, Validator {
+  localValue;
+  @Input() disabled;
+  // The choices for the object's value.
+  @Input() choices;
+  // The schema for this object.
+  // TODO(sll): Validate each choice against the schema.
+  @Input() schema;
+  onChange: (val: unknown) => void = () => {};
+
+  // Implemented as a part of ControlValueAccessor interface.
+  writeValue(value: unknown): void {
+    this.localValue = value;
+  }
+
+  // Implemented as a part of ControlValueAccessor interface.
+  registerOnChange(fn: (val: unknown) => void): void {
+    this.onChange = fn;
+  }
+
+  // Implemented as a part of ControlValueAccessor interface.
+  registerOnTouched(): void {
+  }
+
+  // Implemented as a part of Validator interface.
+  validate(control: AbstractControl): ValidationErrors {
+    return {};
+  }
+
+  updateValue(val: unknown[]): void {
```

Comment:
Should probably validate that the new value is one of the choices given in the schema?

Reply:
Yes, that would be perfect. But I was just trying to get to a stage where it matched what was before. Adding new functions like that might lead to more breakages and sometimes a small thing turns into a big effort for which I felt that didn't have enough time.

Reply:
Hm, this one seems trivial. But if you don't want to do it now, could you please file an issue and link to it here in a TODO? That would be fine too.

Reply:
Done!